<!DOCTYPE html>
<html lang="en">

      
  <div id="readme" class="Box-body readme blob instapaper_body js-code-block-container">
    <article class="markdown-body entry-content p-5" itemprop="text"><table data-table-type="yaml-metadata">
  <thead>
  <tr>
  <th>layout</th>
  <th>title</th>
  <th>zenodo_link</th>
  <th>enable</th>
  <th>questions</th>
  <th>objectives</th>
  <th>key_points</th>
  <th>contributors</th>
  </tr>
  </thead>
  <tbody>
  <tr>
  <td><div>tutorial_hands_on</div></td>
  <td><div>EWAS data analysis of 450k data</div></td>
  <td><div><a href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y" rel="nofollow">https://zenodo.org/record/1251211#.WwREQ1Mvz-Y</a></div></td>
  <td><div>false</div></td>
  <td><div><table>
  <tbody>
  <tr>
  <td><div>What is a epigenome-wide association studies?</div></td>
  <td><div>Why is a EWAS analysis useful?</div></td>
  <td><div>Why 450k data?</div></td>
  </tr>
  </tbody>
</table>
</div></td>
  <td><div><table>
  <tbody>
  <tr>
  <td><div>Learn how to perform reproducible EWAS analysis</div></td>
  </tr>
  </tbody>
</table>
</div></td>
  <td><div><table>
  <tbody>
  <tr>
  <td><div>EWAS analyse changes in dna methylation in cohorts of different individuals effected by diseases</div></td>
  <td><div>DNA methylation changes can be triggered by environment (i.e. sun exposure, diet) and effect the way how our cells works and in long term cause cancer</div></td>
  </tr>
  </tbody>
</table>
</div></td>
  <td><div><table>
  <tbody>
  <tr>
  <td><div>kpbioteam</div></td>
  <td><div>kpoterlowicz</div></td>
  </tr>
  </tbody>
</table>
</div></td>
  </tr>
  </tbody>
</table>

<h1><a id="user-content-introduction" class="anchor" aria-hidden="true" href="#introduction"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Introduction</h1>
<p>{:.no_toc}</p>
<p>The field of cancer genomics has demonstrated the power of massively parallel sequencing techniques to inform on the genes and specific alterations that drive tumor onset and progression. Although large comprehensive sequence data sets continue to be made increasingly available, data analysis remains an ongoing challenge, particularly for laboratories lacking dedicated resources and bioinformatics expertise. To address this, we have provide training based on  Galaxy tools ewas suite that represent many popular algorithms for detecting somatic genetic alterations from cancer genome and exome data.</p>
<p><a target="_blank" rel="noopener noreferrer" href="/galaxyproject/training-material/blob/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/Epigenetic_mechanisms.jpg"><img src="/galaxyproject/training-material/raw/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/Epigenetic_mechanisms.jpg" alt="epimechanism" title="How epigenetics mechanism can effect health (adapted from https://commonfund.nih.gov/epigenomics/figure)" style="max-width:100%;"></a></p>
<p>This exercise uses datasets from the Cell publication by Hugo, Willy, et al., 2015 (DOI: 10.1016/j.cell.2015.07.061). with the goal to identify differentially methylated regions and positions associated with treatment resistant melanomas.
Datasets include Illumina 450K Methylation array performed in melanoma tumors in sample of patients pre and post MAPKi and BRAFi treatment with  different outcomes (sensitive and resistant). For each sample there is raw green and red colour arrays containing the summarized bead information generated bythe Illumina 450k scanner, i.e. there are 8 files in total:</p>
<p>The Infinium Methylation Assay 450k uses two different bead types to detect changes in DNA methylation levels. In the figure we can see M - methylated and U - unmethylated bead types. In our study unmethylated and methylated bead signals are reported as green and red colors respectively.</p>
<p><a target="_blank" rel="noopener noreferrer" href="/galaxyproject/training-material/blob/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/methylationassay.png"><img src="/galaxyproject/training-material/raw/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/methylationassay.png" alt="methassay" title="Infinium Methylation Assay Overview (adapted from https://emea.illumina.com/science/technology/beadarray-technology/infinium-methylation-assay.html?langsel=/gb/)" style="max-width:100%;"></a></p>
<table>
<thead>
<tr>
<th>Accession</th>
<th>Sensitivity</th>
<th>Treatment</th>
</tr>
</thead>
<tbody>
<tr>
<td>GSM1588704</td>
<td>baseline</td>
<td>pre-treatment</td>
</tr>
<tr>
<td>GSM1588705</td>
<td>baseline</td>
<td>pre-treatment</td>
</tr>
<tr>
<td>GSM1588706</td>
<td>resistant</td>
<td>BRAFi</td>
</tr>
<tr>
<td>GSM1588707</td>
<td>resistant</td>
<td>BRAFi</td>
</tr>
</tbody>
</table>
<blockquote>
<h3><a id="user-content-agenda" class="anchor" aria-hidden="true" href="#agenda"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Agenda</h3>
<ol>
<li>TOC
{:toc}</li>
</ol>
</blockquote>
<p>{: .agenda}</p>
<h1><a id="user-content-450k-array-data-upload" class="anchor" aria-hidden="true" href="#450k-array-data-upload"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>450K Array Data Upload</h1>
<p>The first step of EWAS data anylalysis is the raw methylation data loading (intensity information files for each two color micro array)</p>
<blockquote>
<h3><a id="user-content--icon-hands_on--hands-on-data-loading" class="anchor" aria-hidden="true" href="#-icon-hands_on--hands-on-data-loading"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon hands_on %} Hands-on: Data Loading</h3>
<ol>
<li>
<p>Create a new history for this tutorial and give it a proper name</p>
</li>
<li>
<p>Import the following IDAT files from <a href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y" rel="nofollow">Zenodo</a> or from the data library (ask your instructor)</p>
<ul>
<li><code>GSM1588704_8795207135_R01C02_Red.idat</code></li>
<li><code>GSM1588705_8795207119_R05C02_Red.idat</code></li>
<li><code>GSM1588706_8795207135_R02C02_Red.idat</code></li>
<li><code>GSM1588707_8795207119_R06C02_Red.idat</code></li>
<li><code>GSM1588704_8795207135_R01C02_Grn.idat</code></li>
<li><code>GSM1588705_8795207119_R05C02_Grn.idat</code></li>
<li><code>GSM1588706_8795207135_R02C02_Grn.idat</code></li>
<li><code>GSM1588707_8795207119_R06C02_Grn.idat</code></li>
</ul>
<pre><code>
https://zenodo.org/record/1251211/files/GSM1588704_8795207135_R01C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588706_8795207135_R02C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588705_8795207119_R05C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588707_8795207119_R06C02_Red.idat
https://zenodo.org/record/1251211/files/GSM1588704_8795207135_R01C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588706_8795207135_R02C02_Grn.idat
https://zenodo.org/record/1251211/files/GSM1588705_8795207119_R05C02_Grn.idat   
https://zenodo.org/record/1251211/files/GSM1588707_8795207119_R06C02_Grn.idat 
</code></pre>
</li>
</ol>
</blockquote>
<blockquote>
<pre><code>&gt; ### {% icon tip %} Tip: Importing data via links
&gt;
&gt; * Copy the link locations
&gt; * Open the Galaxy Upload Manager
&gt; * Select **Paste/Fetch Data**
&gt; * Paste the link into the text field
&gt; * Press **Start**
{: .tip}

3. Run **minfi read450k** {% icon tool %} with the appropriate selection of {% icon param-files %} *"red channel files"*: &gt; and {% icon param-files %} *"green channel files"*: on datesets uploaded in the previous step
4. Inspect generated set of data
</code></pre>
</blockquote>
<p>{: .hands_on}</p>
<blockquote>
<h3><a id="user-content--icon-question--questions" class="anchor" aria-hidden="true" href="#-icon-question--questions"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon question %} Questions</h3>
<p>How are the Green and Red signals are stored?</p>
<blockquote>
<h3><a id="user-content--icon-solution--solution" class="anchor" aria-hidden="true" href="#-icon-solution--solution"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon solution %} Solution</h3>
<p>Green and Red micro arrays are builded up into <code>RGChannelSet</code>
{: .solution}</p>
</blockquote>
</blockquote>
<p>{: .question}</p>
<h1><a id="user-content-preprocessing-and--quality-assessment" class="anchor" aria-hidden="true" href="#preprocessing-and--quality-assessment"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Preprocessing and  Quality Assessment</h1>
<p>Preprocessing and data quality assurance is an important step in Infinium Methylation Assay analysis.</p>
<p><code>RGChannelSet</code> represents two color data with a green and a red channel and can be converted into methylated and unmethylated signals assigned to <code>MethylSet</code> or into Beta values build in <code>RatioSet</code>. User can convert from <code>RGChannelSet</code> into <code>MethylSet</code> using the <strong>minfi mset</strong> {% icon tool %} or compute Beta values using <strong>minfi set</strong> {% icon tool %}. The <strong>minfi qc</strong> {% icon tool %} tool extracts and plots the quality control data frame with two columns mMed and uMed which are the medians of <code>MethylSet</code> signals (Meth and Unmeth).Comparing them against one another allows user to detect and remove low-quality samples.</p>
<blockquote>
<h3><a id="user-content--icon-hands_on--hands-on-preprocessing" class="anchor" aria-hidden="true" href="#-icon-hands_on--hands-on-preprocessing"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon hands_on %} Hands-on: Preprocessing</h3>
<ol>
<li>Run <strong>minfi mset</strong> {% icon tool %} on the output from minfi_read450K to create <code>MethylSet</code> object</li>
<li>Run <strong>minfi qc</strong> {% icon tool %}  on the 'MethylSet' object created in the previous step to estimate sample-specific quality control. Once it is done click on the {% icon galaxy-eye %} (eye) icon next to the QC filename in your history to view QC image.</li>
<li>Once you have inspected the quality control run <strong>minfi rset</strong> {% icon tool %} on the  'MethylSet'output from minfi_read450K  to convert methylation data from the <code>MethylSet</code> to ratio of the methylated probe intensity and the overall intensity (sum of methylated and unmethylated probe intensities)</li>
<li>Run <strong>minfi maptogenome</strong> {% icon tool %} on the output from the previous step to map ratio data to the genomic positions</li>
</ol>
<blockquote>
<h3><a id="user-content--icon-comment--preprocess-and-normalize-data" class="anchor" aria-hidden="true" href="#-icon-comment--preprocess-and-normalize-data"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon comment %} Preprocess and Normalize data</h3>
<p>If your files require normalisation, you might prefer to use other of preprocessing tools provided in EWAS suite i.e. <strong>minfi ppfun</strong> {% icon tool %} or <strong>minfi ppquantile</strong>  {% icon tool %} look for recomendation at (ref).</p>
</blockquote>
<p>{: .comment}</p>
</blockquote>
<p>{: .hands_on}</p>
<p><a target="_blank" rel="noopener noreferrer" href="/galaxyproject/training-material/blob/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/qcplot.png"><img src="/galaxyproject/training-material/raw/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/qcplot.png" alt="Quality Control" title="Quality control plot" style="max-width:100%;"></a></p>
<h1><a id="user-content-removing-probes-affected-by-genetic-variation" class="anchor" aria-hidden="true" href="#removing-probes-affected-by-genetic-variation"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Removing probes affected by genetic variation</h1>
<p>Incomplete annotation of genetic variations such as single nucleotide polymorphism (SNP) may affect DNA measurements and interfere results from downstream analysis.</p>
<p>Hansen, J. P. 2014 highly recommende to remove the probes that contain either a SNP at the metylated loci interrogation or at the single nucleotide extension</p>
<blockquote>
<h3><a id="user-content--icon-hands_on--hands-on-removing-probes-affected-by-genetic-variation" class="anchor" aria-hidden="true" href="#-icon-hands_on--hands-on-removing-probes-affected-by-genetic-variation"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon hands_on %} Hands-on: Removing probes affected by genetic variation</h3>
<ol>
<li>Run <strong>minfi dropsnp</strong> {% icon tool %}  on the output of minfi_maptogenome to remove the probes that contain either a SNP at the metylated loci interrogation or at the single nucleotide extension
{: .hands_on}</li>
</ol>
</blockquote>
<h1><a id="user-content-dmps-and-dmrs-identification" class="anchor" aria-hidden="true" href="#dmps-and-dmrs-identification"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>DMPs and DMRs Identification</h1>
<p>The main goal of the EWAS suite is to simplify the way differentially methylated loci sites are detected. The EWAS suite contains <strong>minfi dmp</strong> {% icon tool %} tool detecting differentially methylated positions (DMPs) with respect to a phenotype covariate, and more complex <strong>minfi dmr</strong> {% icon tool %} solution for finding differentially methylated regions (DMRs). Genomic regions that are differentially methylated between two conditions can be tracked using a bumphunting algorithm. The algorithm first implements a t-statistic at each methylated loci location, with optional smoothing, then groups probes into clusters with a maximum location gap and a cutoff size to refer the lowest possible value of genomic profile hunted by our tool.</p>
<p>In order the next we will need additional file that annotate illumina450K with genomic features, which we can obtain from UCSC.</p>
<blockquote>
<h3><a id="user-content--icon-hands_on--hands-on-data-upload-from-ucsc" class="anchor" aria-hidden="true" href="#-icon-hands_on--hands-on-data-upload-from-ucsc"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon hands_on %} Hands-on: Data upload from UCSC</h3>
<ol>
<li>
<p>Search for <code>UCSC Main</code> in the tool search bar (top left)</p>
</li>
<li>
<p>Click on <code>UCSC Main</code> {% icon tool %}</p>
<p>You will be taken to the <strong>UCSC table browser</strong>, which looks something like this:</p>
<p><a target="_blank" rel="noopener noreferrer" href="/galaxyproject/training-material/blob/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/UCSC_genome_browser.png"><img src="/galaxyproject/training-material/raw/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/UCSC_genome_browser.png" alt="UCSC table browser interface" style="max-width:100%;"></a></p>
</li>
<li>
<p>Set the following options:</p>
<ul>
<li><em>"clade"</em>: <code>Mammal</code></li>
<li><em>"genome"</em>: <code>Human</code></li>
<li><em>"assembly"</em>: <code>Feb. 2009 (GRCh37/hg19)</code></li>
<li><em>"group"</em>: <code>Regulation</code></li>
<li><em>"track"</em>: <code>HAIB Methyl450</code></li>
<li><em>"table"</em>: <code>GM12878 (wgEncodeHaibMethyl450Gm12878SitesRep1)</code></li>
<li><em>"region"</em>: <code>genome</code></li>
<li><em>"output format"</em>: <code>GTF - gene transfer (limited)</code></li>
<li><em>"Send output to"</em>: <code>Galaxy</code> (only)</li>
</ul>
</li>
<li>
<p>Click on the <strong>get output</strong> button</p>
<p>You will see the next screen:</p>
<p><a target="_blank" rel="noopener noreferrer" href="/galaxyproject/training-material/blob/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/OutputHAIB450K_to_Galaxy.png"><img src="/galaxyproject/training-material/raw/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/OutputHAIB450K_to_Galaxy.png" alt="Output settings" style="max-width:100%;"></a></p>
</li>
<li>
<p>Click on the <strong>Send Query to Galaxy</strong> button</p>
</li>
<li>
<p>Wait for the upload to finish</p>
</li>
</ol>
</blockquote>
<p>{: .hands_on}</p>
<blockquote>
<h3><a id="user-content--icon-hands_on--hands-on-dmps-and-dmrs-identification" class="anchor" aria-hidden="true" href="#-icon-hands_on--hands-on-dmps-and-dmrs-identification"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon hands_on %} Hands-on: DMPs and DMRs Identification</h3>
<ol>
<li>
<p>Import <code>phenotypeTable.txt</code> from <a href="https://zenodo.org/record/1251211#.WwREQ1Mvz-Y" rel="nofollow">Zenodo</a> or data library
{% icon tip %} Tip: Importing data via links</p>
<pre><code>https://zenodo.org/record/1251211/files/phenotypeTable.txt
</code></pre>
</li>
<li>
<p>Run <strong>minfi dmp</strong> {% icon tool %} on the output of the minfi dropsnp tool adjusting the following parameters</p>
<ul>
<li>
<p>{% icon param-file %} <em>"Input set"</em>:<code>GenomicRatioSet</code></p>
</li>
<li>
<p>{% icon param-file %} <em>"Phenotype Table"</em>:<code>phenotypeTable.txt</code></p>
</li>
<li>
<p><em>"Phenotype Type"</em>:<code>categorical</code></p>
</li>
<li>
<p><em>"qCutoff Size"</em>:<code>0.5</code></p>
<p>DMPs with an FDR q-value greater than this will not be returned</p>
</li>
<li>
<p><em>"Variance Shrinkage"</em>:<code> TRUE</code></p>
<p>This is recommended when sample sizes are small &lt;10</p>
</li>
<li>
<p><em>"Genome Table"</em>: <code>wgEncodeHaibMethyl450 ... </code></p>
</li>
</ul>
</li>
<li>
<p>Run <strong>minfi dmr</strong> {% icon tool %} on the output of minfi_dmp adjusting the following parameters</p>
<ul>
<li>{% icon param-file %} <em>"Input set"</em>:<code>GenomicRatioSet</code></li>
<li>{% icon param-file %} <em>"Phenotype Table"</em>:<code>phenotypeTable.txt</code></li>
<li><em>"factor1"</em>: <code>sensitive</code></li>
<li><em>"factor2"</em>: <code>resistant</code></li>
<li><em>"maxGap Size"</em>:<code>250</code></li>
<li><em>"coef Size"</em>:<code>2</code></li>
<li><em>"Cutoff Size"</em>:<code>0.1</code></li>
<li><em>"nullMethod"</em>:<code>permutation</code></li>
<li><em>"verbose"</em>:<code>TRUE</code></li>
</ul>
</li>
<li>
<p>Click on the <strong>minfi dmr</strong> {% icon tool %} output in your history to expand it</p>
</li>
<li>
<p>Click on the {% icon galaxy-pencil %} (pencil) icon displayed in your dataset in the history</p>
</li>
<li>
<p>Set Database/Build to <code>Human Feb. 2009 (GRCh37/hg19) (hg19)</code></p>
</li>
<li>
<p>Press <strong>Save</strong></p>
</li>
<li>
<p>Click on <code>display at UCSC</code> towards the bottom of the history item
This will launch UCSC Genome Browser with your Custom Track
{: .hands_on}</p>
</li>
</ol>
</blockquote>
<p><a target="_blank" rel="noopener noreferrer" href="/galaxyproject/training-material/blob/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/ucsc.png"><img src="/galaxyproject/training-material/raw/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/ucsc.png" alt="Display at UCSC" title="UCSC genome track showing differentialy methylated regions located on chromosome 6" style="max-width:100%;"></a></p>
<blockquote>
<h3><a id="user-content--icon-comment--phenotype-table" class="anchor" aria-hidden="true" href="#-icon-comment--phenotype-table"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon comment %} Phenotype table</h3>
<p>Phenotype table can be in diffrent size with diffrent arguments only second column is required to contain phenotype covariate information for each sample.
{: .comment}</p>
</blockquote>
<blockquote>
<h3><a id="user-content--icon-question--questions-1" class="anchor" aria-hidden="true" href="#-icon-question--questions-1"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon question %} Questions</h3>
<p>How are we define phenotype covariate?</p>
<blockquote>
<h3><a id="user-content--icon-solution--solution-1" class="anchor" aria-hidden="true" href="#-icon-solution--solution-1"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon solution %} Solution</h3>
<p>Phenotype covariate is the set of observable characteristics of an individual resulting from the gene-environment interactions
{: .solution}
{: .question}</p>
</blockquote>
</blockquote>
<h1><a id="user-content-annotation-and-visualization" class="anchor" aria-hidden="true" href="#annotation-and-visualization"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Annotation and Visualization</h1>
<p>In addition to downstream analysis users can annotate the differentially methylated loci to the promoter regions of genes with gene function description, and relationships between these concepts.</p>
<blockquote>
<h3><a id="user-content--icon-hands_on--hands-on--annotate-differentially-methylated-position" class="anchor" aria-hidden="true" href="#-icon-hands_on--hands-on--annotate-differentially-methylated-position"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>{% icon hands_on %} Hands-on:  Annotate Differentially Methylated Position</h3>
<ol>
<li>Run <strong>chipeakanno annopeaks</strong> {% icon tool %}on the output of minfi_dmp with the following parameters</li>
</ol>
<ul>
<li>
<p>{% icon param-file %} <em>"Differentialy methylated data"</em>: output of <strong>minfi dmp</strong> {% icon tool %}</p>
</li>
<li>
<p><em>"bindingType"</em>: <code>StartSite</code></p>
</li>
<li>
<p><em>"bindingRegionStart"</em>:<code>-5000</code></p>
</li>
<li>
<p><em>"bindingRegionEnd"</em>:<code>3000</code></p>
</li>
<li>
<p><em>"Additional Column of Score"</em>:<code>8</code></p>
<p>Position of column of score optional value if it is required</p>
</li>
</ul>
<ol start="2">
<li><strong>Cut</strong> {% icon tool %} on the previous output adjusting the following parameters to cut "gene_name" column from table of annotated peaks and then get a list of geenes</li>
</ol>
<ul>
<li><em>"Cut columns"</em>: <code>c16</code></li>
<li><em>"Delimited by"</em>: <code>Tab</code></li>
<li>{% icon param-file %} <em>"From"</em>: output of <strong>chipeakanno annopeaks</strong> {% icon tool %}</li>
</ul>
<ol start="3">
<li><strong>Remove beginning</strong> {% icon tool %} of <code>Gene List</code> with the following parameters</li>
</ol>
<ul>
<li><em>"Remove first"</em>: <code>1</code></li>
<li>{% icon param-file %} <em>"from"</em>: output of <strong>Cut</strong> {% icon tool %}</li>
</ul>
<ol start="4">
<li>Run  <strong>clusterProfiler bitr</strong> {% icon tool %} on the previous output adjusting the following parameters to convert the list of genes to list of entrez ID</li>
</ol>
<ul>
<li><em>"Input Type Gene ID"</em>: <code>SYMBOL</code></li>
<li><em>"Output Type Gene ID"</em>: <code>ENTREZID</code></li>
</ul>
<ol start="5">
<li>Use the outpu of the  clusterProfiler bitr {% icon tool %} to run a GO Enrichment Analysis using <strong>clusterProfiler go</strong></li>
</ol>
</blockquote>
<p>{: .hands_on}</p>
<p><a target="_blank" rel="noopener noreferrer" href="/galaxyproject/training-material/blob/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/funcann.jpg"><img src="/galaxyproject/training-material/raw/710c7e9a209589e33106f832775e5044f74792e2/topics/epigenetics/images/funcann.jpg" alt="Functional annotations" title="Results of GO enrichments analysis for DMPs" style="max-width:100%;"></a></p>
<table>
<thead>
<tr>
<th>ID</th>
<th>Description</th>
<th>pvalue</th>
<th>qvalue</th>
<th>geneID</th>
<th>Count</th>
</tr>
</thead>
<tbody>
<tr>
<td>GO:0048732</td>
<td>gland development</td>
<td>1.38E-58</td>
<td>4.23E-55</td>
<td>PTGS2 / KCNC1 / FZD1 /SLC22A18 /SLC22A3 (...)</td>
<td>372</td>
</tr>
<tr>
<td>GO:1901652</td>
<td>response to peptide</td>
<td>3.99E-57</td>
<td>8.13E-54</td>
<td>SULF1/ LAMA5/ MED1 /CFLAR/ MSX2 (...)</td>
<td>359</td>
</tr>
<tr>
<td>GO:0048545</td>
<td>response to steroid hormone</td>
<td>1.38EE-54</td>
<td>2.11E-51</td>
<td>HDAC9/ RAB10/ CFLAR/ WDTC1 (...)</td>
<td>394</td>
</tr>
</tbody>
</table>
<h1><a id="user-content-conclusion" class="anchor" aria-hidden="true" href="#conclusion"><svg class="octicon octicon-link" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Conclusion</h1>
<p>{:.no_toc}</p>
<p>Epigenetic aberrations which involve DNA modifications give researchers an interest to identify novel non-genetic factors responsible for complex human phenotypes such as height, weight, and disease. To identify methylation changes researchers need to perform complicated and  time consuming computational analysis. Here, the EWAS suite becomes a solution for this inconvenience and provides a simplified downstream analysis available as ready to run pipline in supplementary materials.</p>
</article>
  </div>

    </div>

  

  <details class="details-reset details-overlay details-overlay-dark">
    <summary data-hotkey="l" aria-label="Jump to line"></summary>
    <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast linejump" aria-label="Jump to line">
      <!-- '"` --><!-- </textarea></xmp> --></option></form><form class="js-jump-to-line-form Box-body d-flex" action="" accept-charset="UTF-8" method="get"><input name="utf8" type="hidden" value="&#x2713;" />
        <input class="form-control flex-auto mr-3 linejump-input js-jump-to-line-field" type="text" placeholder="Jump to line&hellip;" aria-label="Jump to line" autofocus>
        <button type="submit" class="btn" data-close-dialog>Go</button>
</form>    </details-dialog>
  </details>



  </div>
  <div class="modal-backdrop js-touch-events"></div>
</div>

    </main>
  </div>
  

  </div>

        
<div class="footer container-lg width-full px-3" role="contentinfo">
  <div class="position-relative d-flex flex-justify-between pt-6 pb-2 mt-6 f6 text-gray border-top border-gray-light ">
    <ul class="list-style-none d-flex flex-wrap ">
      <li class="mr-3">&copy; 2019 <span title="0.24491s from unicorn-5d74d86b65-jqgnr">GitHub</span>, Inc.</li>
        <li class="mr-3"><a data-ga-click="Footer, go to terms, text:terms" href="https://github.com/site/terms">Terms</a></li>
        <li class="mr-3"><a data-ga-click="Footer, go to privacy, text:privacy" href="https://github.com/site/privacy">Privacy</a></li>
        <li class="mr-3"><a data-ga-click="Footer, go to security, text:security" href="https://github.com/security">Security</a></li>
        <li class="mr-3"><a href="https://githubstatus.com/" data-ga-click="Footer, go to status, text:status">Status</a></li>
        <li><a data-ga-click="Footer, go to help, text:help" href="https://help.github.com">Help</a></li>
    </ul>

    <a aria-label="Homepage" title="GitHub" class="footer-octicon mx-lg-4" href="https://github.com">
      <svg height="24" class="octicon octicon-mark-github" viewBox="0 0 16 16" version="1.1" width="24" aria-hidden="true"><path fill-rule="evenodd" d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0 0 16 8c0-4.42-3.58-8-8-8z"/></svg>
</a>
   <ul class="list-style-none d-flex flex-wrap ">
        <li class="mr-3"><a data-ga-click="Footer, go to contact, text:contact" href="https://github.com/contact">Contact GitHub</a></li>
        <li class="mr-3"><a href="https://github.com/pricing" data-ga-click="Footer, go to Pricing, text:Pricing">Pricing</a></li>
      <li class="mr-3"><a href="https://developer.github.com" data-ga-click="Footer, go to api, text:api">API</a></li>
      <li class="mr-3"><a href="https://training.github.com" data-ga-click="Footer, go to training, text:training">Training</a></li>
        <li class="mr-3"><a href="https://github.blog" data-ga-click="Footer, go to blog, text:blog">Blog</a></li>
        <li><a data-ga-click="Footer, go to about, text:about" href="https://github.com/about">About</a></li>

    </ul>
  </div>
  <div class="d-flex flex-justify-center pb-6">
    <span class="f6 text-gray-light"></span>
  </div>
</div>



  <div id="ajax-error-message" class="ajax-error-message flash flash-error">
    <svg class="octicon octicon-alert" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"/></svg>
    <button type="button" class="flash-close js-ajax-error-dismiss" aria-label="Dismiss error">
      <svg class="octicon octicon-x" viewBox="0 0 12 16" version="1.1" width="12" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48L7.48 8z"/></svg>
    </button>
    You can’t perform that action at this time.
  </div>


    
    <script crossorigin="anonymous" integrity="sha512-m1DeH+jLA0IS/zzy50Kv0yBpZJDQlBkcACJzTGJUq30SG1eo5vztaQg1x1KV1L3XGePC1HOdKU1fu7Ktr+5nsg==" type="application/javascript" src="https://github.githubassets.com/assets/frameworks-3c669c6d.js"></script>
    
    <script crossorigin="anonymous" async="async" integrity="sha512-5fkE5HlHUvjDCSLq2ACLerBiuS+N3x36xiX0Dld1w8J0+IpjD1ddnmRd806jtn+QoMMGkLFSUMXVvMN4JUkZSw==" type="application/javascript" src="https://github.githubassets.com/assets/github-bootstrap-7f581dd4.js"></script>
    
    
    
  <div class="js-stale-session-flash stale-session-flash flash flash-warn flash-banner" hidden
    >
    <svg class="octicon octicon-alert" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z"/></svg>
    <span class="signed-in-tab-flash">You signed in with another tab or window. <a href="">Reload</a> to refresh your session.</span>
    <span class="signed-out-tab-flash">You signed out in another tab or window. <a href="">Reload</a> to refresh your session.</span>
  </div>
  <template id="site-details-dialog">
  <details class="details-reset details-overlay details-overlay-dark lh-default text-gray-dark" open>
    <summary aria-haspopup="dialog" aria-label="Close dialog"></summary>
    <details-dialog class="Box Box--overlay d-flex flex-column anim-fade-in fast">
      <button class="Box-btn-octicon m-0 btn-octicon position-absolute right-0 top-0" type="button" aria-label="Close dialog" data-close-dialog>
        <svg class="octicon octicon-x" viewBox="0 0 12 16" version="1.1" width="12" height="16" aria-hidden="true"><path fill-rule="evenodd" d="M7.48 8l3.75 3.75-1.48 1.48L6 9.48l-3.75 3.75-1.48-1.48L4.52 8 .77 4.25l1.48-1.48L6 6.52l3.75-3.75 1.48 1.48L7.48 8z"/></svg>
      </button>
      <div class="octocat-spinner my-6 js-details-dialog-spinner"></div>
    </details-dialog>
  </details>
</template>

  <div class="Popover js-hovercard-content position-absolute" style="display: none; outline: none;" tabindex="0">
  <div class="Popover-message Popover-message--bottom-left Popover-message--large Box box-shadow-large" style="width:360px;">
  </div>
</div>

  <div aria-live="polite" class="js-global-screen-reader-notice sr-only"></div>

  </body>
</html>

